### PR TITLE
Improve Subgraph querying for SDK-Core & SDK-Redux

### PR DESCRIPTION
**Pull request by kasparkallas**
Why?
Previous solution was too limited. Now all Subgraph filters and orderings are enabled. Also single entities are queryable by ID.

How?
SDK-Redux now exports `sfSubgraph` slice which supports the new query functionality.

SDK-Core has QueryHandlers which is a generic base class for creating new queries with minimal effort. The code is re-used by SDK-Redux so the code on SDK-Redux level is decreased.

What more?
* Needs some clean up (TODO-s & organization & documentation).
* Haven't actually figured out what's the best way to expose the querying from SDK-Core yet. Currently, the QueryHandlers are exposed which are used by SDK-Redux.
* Caching logic not completely finished.
* Not all the event queries are transfered into the new format.

### CODE REVIEW

**Comment by kasparkallas** [comment]
Checking as _not a draft_ to get a temporary package build.

File: packages/sdk-core/src/subgraph/normalizeSubgraphFilter.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            return value.toLowerCase();
+        }
+    }
+
+    if (Array.isArray(value)) {
+        return value.map((x) => lowerCaseIfAddress(x));
+    }
+
+    return value;
+};
```
Comment (hellwolf) [inline packages/sdk-core/src/subgraph/normalizeSubgraphFilter.ts:36]: relevantAddressProvider.ts is empty

File: packages/sdk-redux/package.json [hunk 2]
```diff
@@ -34,7 +34,8 @@
     },
     "dependencies": {
         "ethers": "^5.5.1",
-        "lodash": "^4.17.21"
+        "lodash": "^4.17.21",
+        "graphql-request": "^3.7.0"
```
Comment (hellwolf) [inline packages/sdk-redux/package.json:38]: if we have time, we can revisit our monorepo hoisting strategy to resolve this type of issue in the future https://classic.yarnpkg.com/blog/2018/02/15/nohoist/

File: packages/sdk-redux/src/redux-slices/argTypes.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * Information about the transaction to locate it.
+ * Information about the transaction to locate it.BaseQu
```
Comment (hellwolf) [inline packages/sdk-redux/src/redux-slices/argTypes.ts:2]: typo

File: packages/sdk-redux/src/redux-slices/rtk-query/cacheTime.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export enum CacheTime {
+    None = 0,
+    OneMinute = 60,
+    TwoMinutes = 120,
+    ThreeMinutes = 180,
+    FiveMinutes = 180,
```
Comment (hellwolf) [inline packages/sdk-redux/src/redux-slices/rtk-query/cacheTime.ts:6]: typo

File: packages/sdk-redux/src/redux-slices/rtk-query/getSerializeQueryArgs.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                          return acc;
+                      }, {})
+                : normalizeValue(value)
+        )})`;
+    };
+
+// TODO(KK): The logic below is duplicated in SDK-Core. Think of a way of sharing it.
+
+// NOTE: Regex taken from Ethers.
+const isAddressRegex = /^(0x)?[0-9a-fA-F]{40}$/;
```
Comment (hellwolf) [inline packages/sdk-redux/src/redux-slices/rtk-query/getSerializeQueryArgs.ts:25]: reuse it?

**Review by hellwolf — COMMENTED** [review]
Looks good. @0xdavinchee please go forward with the rest of the reviews

**Review by 0xdavinchee — APPROVED** [review]
comments addressed after review last week

File: examples/sdk-redux-react-typecript/src/features/generic-entity-queries/Accounts.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    FormGroup,
+    Radio,
+    RadioGroup,
+    TextField,
+} from "@mui/material";
+import { SignerContext } from "../../SignerContext";
+import { sfSubgraph } from "../../redux/store";
+import { GridSortModel } from "@mui/x-data-grid";
+import { GenericDataGrid } from "./GenericDataGrid";
+import { Account_OrderBy } from "@superfluid-finance/sdk-core/dist/module/subgraph/schema.generated";
```
[the remark this answers is not in the document]
Comment (kasparkallas) [reply examples/sdk-redux-react-typecript/src/features/generic-entity-queries/Accounts.tsx]: Generated Subgraph schema is now exported from SDK-Core as 1st class citizen. This means we export these utility types:
```
export type Maybe<T> = T | undefined;
export type InputMaybe<T> = T | undefined | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export interface Scalars {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  BigDecimal: string;
  BigInt: string;
  Bytes: string;
}```

File: packages/sdk-redux/src/redux-slices/rtk-query/getSerializeQueryArgs.ts (hunk 5 above)
Comment (kasparkallas) [reply packages/sdk-redux/src/redux-slices/rtk-query/getSerializeQueryArgs.ts:25]: I'll leave it as it is for now as I don't have a very clear idea of the best way of re-using it. I'll tackle it next time I come across it.

**Comment by hellwolf** [comment]
Good to go? @kasparkallas ?

**Pull request merged**